File: 167-two-sum-ii-input-array-is-sorted/167-two-sum-ii-input-array-is-sorted.cs
public class Solution {
    
    /// <summary>
    /// 
    /// IDEA:
    /// There will always be one solution (never no solution)
    /// Non-decreasing means the numbers get larger as we move from left to right.
    /// Use two pointers, one starting at either end of the array, that move
    /// towards the center of the array. If the sum of the values at the two
    /// pointers is greater than the target int, move the right pionter to
    /// the left. If the sum is less than the target, move the left pointer
    /// to the right. If the sum equals the target, return a new array
    /// containing the indices of the two pointers (each should be incremmented)
    /// by one before being added to the array due to the 1-indexing requirement.
    /// 
    /// </summary>
    public int[] TwoSum(int[] numbers, int target) {
        
        int left = 0;
        int right = numbers.Length - 1;
        int sum = numbers[left] + numbers[right];
        
        while (sum != target) {

            if (sum < target) {
                left++;
            } else { // sum > target
                right--;
            }
            sum = numbers[left] + numbers[right];
        }

        int[] indices = {left + 1, right + 1};
        return indices;
    }
}

File: LeetCode/0167-two-sum-ii-input-array-is-sorted/167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left = 0
        right = len(numbers) - 1
        sum = numbers[left] + numbers[right]

        while sum != target:
            if sum < target:
                left += 1
            if sum > target:
                right -= 1
            sum = numbers[left] + numbers[right]
        
        return [left + 1, right + 1]

# OUTLINE
# > Note: Input array is sorted in ascending order. Ex. [1, 2, 3, 4, ...]
# > Note: Output is to be 1-indexed. NOT zero indexed!
# IDEA:
# > Create two pointers with which to loop over the array: one starting at the beginning and the other at the end. Then loop over the array until the values at the two pointers sum to the target value. On each iteration of the loop, if currentSum is less than target, increment the left pointer by one; if currentSum is greater than the target, decrement the right pointer by one.
# PSEUDO CODE
# Instantiate left pointer and right pointer: left and right
# Instantiate `sum` to the sum of numbers[left] + numbers[right]
# Loop: While sum != target,
#     If sum < target,
#         left++
#     If sum > target,
#         right--
#     Update `sum` as left + right
# Instantiate array containing `left + 1` and `right + 1`
# Return array